We get a build failure for `hsc2hs-0.68.10` under GHC 8.2 and 8.0 here:
https://github.com/haskell-hvr/windns/actions/runs/6909745495/job/18801646012#step:4:22
```
Building executable 'hsc2hs' for hsc2hs-0.68.10..
...
[12 of 12] Compiling Main             ( src\Main.hs, dist\build\hsc2hs\hsc2hs-tmp\Main.o )
In file included from C:\ghcup\ghc\8.2.2\lib\base-4.10.1.0\include/HsBase.h:29,

Error: cabal-3.10.2.0.exe: Failed to build exe:hsc2hs from hsc2hs-0.68.10
(which is required by windns-0.1.0.1). See the build log above for details.

                 from cbits\utils.c:9:0: error: 
C:\ghcup\ghc\8.2.2\lib\base-4.10.1.0\include/HsBase.h:545:15: warning: '__p__environ' redeclared without dllimport attribute: previous dllimport ignored [-Wattributes]
  545 | extern char** environ;
      |               ^~~~~~~
Linking dist\build\hsc2hs\hsc2hs.exe ...
dist\build\hsc2hs\hsc2hs-tmp\cbits\utils.o:utils.c:(.text+0x33): undefined reference to `__stdio_common_vswprintf_s'
C:\sr\ghc-8.2.2\process-1.6.18.0-658f55d[27](https://github.com/haskell-hvr/windns/actions/runs/6909745495/job/18801646012#step:4:28)d6ee087180e4eea644a00c1656ef[33](https://github.com/haskell-hvr/windns/actions/runs/6909745495/job/18801646012#step:4:34)d\lib/libHSprocess-1.6.18.0-658f55d27d6ee087180e4eea644a00c1656ef33d.a(runProcess.o):runProcess.c:(.text+0x1[37](https://github.com/haskell-hvr/windns/actions/runs/6909745495/job/18801646012#step:4:38)): undefined reference to `__stdio_common_vswprintf_s'
collect2.exe: error: ld returned 1 exit status
```
I already reported this issue:
- https://github.com/haskell/hsc2hs/issues/81

I'll proceed and shut off CI for GHC < 8.4 here.

--- a/.github/workflows/win.yml
+++ b/.github/workflows/win.yml
@@ -12,19 +12,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ['9.8', '9.6', '9.4', '9.2', '9.0', '8.10', '8.8', '8.6', '8.4', '8.2', '8.0']
+        ghc: ['9.8', '9.6', '9.4', '9.2', '9.0', '8.10', '8.8', '8.6', '8.4']
     steps:
     - uses: actions/checkout@v4
     - uses: haskell-actions/setup@v2
       with:
         ghc-version: ${{ matrix.ghc }}
-
-    - name: "Work around haskell/hsc2hs#81: hsc2hs-0.68.10 fails to build with GHC < 8.4"
-      if:  matrix.ghc < 8.4
-      run: |
-        echo "constraints: any.hsc2hs < 0.68.10" > cabal.project.local
-      shell: bash
-
     - run: cabal build
     - run: cabal haddock
     - run: cabal check

--- a/.github/workflows/win.yml
+++ b/.github/workflows/win.yml
@@ -18,6 +18,13 @@ jobs:
     - uses: haskell-actions/setup@v2
       with:
         ghc-version: ${{ matrix.ghc }}
+
+    - name: "Work around haskell/hsc2hs#81: hsc2hs-0.68.10 fails to build with GHC < 8.4"
+      if:  matrix.ghc < 8.4
+      run: |
+        echo "constraints: any.hsc2hs < 0.68.10" > cabal.project.local
+      shell: bash
+
     - run: cabal build
     - run: cabal haddock
     - run: cabal check

